### PR TITLE
fix: burn tax split bug

### PR DESCRIPTION
**Pull request by inon-man**
## Summary of changes
This PR is a bug fix for the Burn Tax Split feature.

The current Burn Tax Split feature does not transfer the community pool amount from the fee collector to the treasury, so the total supply and remaining amount of the fee collector are not accurately calculated.

For example:
1. Where
total supply = 100
fee collector = 100
burn tax rate = 0.4
2. In the AnteHandler
split amount = fee collector * burn tax rate = 40
community pool += split amount // fee collector sends 40 to community pool
burn += fee collector - split amount // fee collector sends 60 to burn account
fee collector = 0
total supply = 40

This PR also included removal of TaxPowerSplitHeight as they are no longer needed in v2.x

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [x] Wrote tests
- [ ] Updated API documentation (client/lcd/swagger-ui/swagger.yaml)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

**Comment by nghuyenthevinh2000** [comment]
panic_man

**Review by ZaradarBH — APPROVED** [review]
LGTM :)

File: custom/auth/ante/tax_test.go [hunk 1]
```diff
@@ -384,359 +384,3 @@ func (suite *AnteTestSuite) TestEnsureMempoolFeesExec() {
 	_, err = antehandler(suite.ctx, tx, false)
 	suite.Require().NoError(err, "Decorator should not have errored on fee higher than local gasPrice")
 }
-
-func (suite *AnteTestSuite) TestEnsureMempoolFeesSendLunaTax() {
```
Comment (nghuyenthevinh2000) [inline custom/auth/ante/tax_test.go:388]: why do you remove a bunch of test here?

Comment (inon-man) [reply custom/auth/ante/tax_test.go:388]: Those tests were for old burn tax (100%) with TaxPowerUpgradeHeight

Comment (nghuyenthevinh2000) [reply custom/auth/ante/tax_test.go:388]: I think we should update tests accordingly

Comment (inon-man) [reply custom/auth/ante/tax_test.go:388]: Okay. I will re-add TaxPowerUpgradeHeight then

Comment (inon-man) [reply custom/auth/ante/tax_test.go]: @nghuyenthevinh2000 Please resolve this thread it if you have no issue

**Review by ZaradarBH — APPROVED** [review]
FML your fast! Beat me by like 3 mins. LGTM buddy, I think this is the cleanest solution :)

**Review by ZaradarBH — APPROVED** [review]
LGTM :)

**Review by nghuyenthevinh2000 — APPROVED** [review]

**Pull request merged**